### ISSUE TITLE
Improve diagnostics module documentation clarity

### ISSUE THREAD

**Issue by anjor** [anjor/gandalf#54]
## Background
PR #51 code reviews identified several documentation issues in the diagnostics module that could confuse users or lead to misinterpretation of results. These are not bugs but clarity improvements.

### 1. Parseval Tolerance Not Explained (Moderate)
**Location**: `tests/test_diagnostics.py` (multiple tests with 10% tolerance)

**Current situation**:
```python
assert rel_error < 0.1  # 10% tolerance
```

Tests allow 10% error for Parseval's theorem (∫E(k)dk ≈ E_total). The reviews questioned: "Is this expected? Typically Parseval should hold to ~1e-6 for spectral methods."

**Why 10% is acceptable**:
- Coarse binning (32 bins for 64³ grid) causes discretization error
- Shell averaging introduces additional error
- Physics is correct; error is from binning algorithm
- Finer binning reduces error (but increases computation)

**Action needed**: Add explanation to test docstrings and/or diagnostics.py module docstring:
```python
"""
Note on Parseval's Theorem Accuracy
-----------------------------------
Energy spectra integrate to total energy within ~10% for default binning (32 bins).
This error comes from:
1. Discrete shell averaging (binning artifacts)
2. Coarse k-space sampling (32 bins for 64³ grid)
3. Edge effects at high-k (Nyquist frequency)

Using finer binning (n_bins=64) reduces error to ~1-2%, at cost of increased computation.
The physics and implementation are correct; the 10% is a binning discretization effect.
"""
```

---

### 2. Energy Formula Comments Inconsistent (Minor)
**Location**: `diagnostics.py:115-124` (and similar locations)

**Current comment**:
```python
# E = (1/2) * |∇φ|² + (1/2) * |∇A∥|²
#   = (1/2) * k²|φ|² + (1/2) * k²|A∥|²
```

**Actual code**:
```python
energy_density = 0.5 * k_perp_squared * (...)
```

**Problem**: Comment says k² but code uses k_perp_squared. This is physically correct for RMHD (energy is in perpendicular gradients only) but the comment is misleading.

**Fix**: Update comments to match implementation:
```python
# E = (1/2) * |∇⊥φ|² + (1/2) * |∇⊥A∥|²  (RMHD: perpendicular gradients only)
#   = (1/2) * k⊥²|φ|² + (1/2) * k⊥²|A∥|²
```

**Note**: This is correct physics - parallel derivatives don't contribute to Elsasser energy in RMHD.

---

### 3. Normalization Differences Not Documented (Moderate)
**Location**: `diagnostics.py` (all three spectrum functions)

**Current situation**:
Different spectrum types use different normalizations:
- 1D: `E_k / (N_total * dk)`
- Perpendicular: `E_k / (N_total * dk_perp)`
- Parallel: `E_k / N_perp` (no dk normalization)

This is intentional but not explained, which could confuse users verifying energy conservation.

**Action needed**: Add explicit note in parallel spectrum docstring:
```python
def energy_spectrum_parallel(state: KRMHDState, n_bins: Optional[int] = None) -> Tuple[Array, Array]:
    """
    ...
    
    Note on Normalization
    ---------------------
    This function uses different normalization than energy_spectrum_1d():
    - 1D spectrum: E(k) / (N_total * dk)  → Integrates to E_total when summed over all k
    - Parallel spectrum: E(kz) / N_perp   → Different integration measure for 1D slice
    
    To verify energy conservation with parallel spectrum:
        np.sum(E_parallel) * dkz ≈ E_total  (within binning error)
    
    The perpendicular modes (kx, ky) are already integrated out in this spectrum.
    """
```

---

### 4. Division by Zero Guard Needed (Minor)
**Location**: `diagnostics.py:153` (and similar in other spectrum functions)

**Current code**:
```python
dk = k_bins[1] - k_bins[0]
E_k = E_k / (N_total * dk)
```

**Problem**: If `n_bins=1` (edge case), dk could be zero or undefined.

**Fix**: Add defensive guard:
```python
dk = jnp.maximum(k_bins[1] - k_bins[0], 1e-10)  # Prevent division by zero
E_k = E_k / (N_total * dk)
```

---

## Tasks
- [ ] Add Parseval tolerance explanation to diagnostics.py module docstring
- [ ] Update energy formula comments to use k⊥² notation consistently
- [ ] Document normalization differences in parallel spectrum docstring
- [ ] Add division by zero guards in spectrum normalization
- [ ] Add examples showing how to verify energy conservation with each spectrum type

## Acceptance Criteria
- Users understand why Parseval holds to 10% (not a bug)
- Energy formula comments match code (k⊥² not k²)
- Normalization differences are explained
- No division by zero errors with edge case inputs

## Priority
**Low** - Documentation clarity, not functional bugs

## Related
- PR #51 (diagnostics implementation)
- Issue #52 (performance optimization)
- Issue #36 (numerical error budget for Poisson bracket - similar documentation pattern)

**Comment by anjor** [comment]
**Partial Progress Update:**

Some improvements from this issue have been implemented:

✅ **Division by zero guard** (Task 4) - Added in `diagnostics.py:187`:
```python
dk = jnp.maximum(k_bins[1] - k_bins[0], 1e-10)  # Prevent division by zero
```

✅ **Energy formula comments** (Task 2) - Comments now clarify "perpendicular derivatives only for RMHD":
```python
# Compute energy density (perpendicular derivatives only for RMHD)
k_perp_squared = kx_3d**2 + ky_3d**2
```

**Still TODO:**
- [ ] Add Parseval tolerance explanation to module docstring (Task 1)
- [ ] Document normalization differences in parallel spectrum docstring (Task 3)
- [ ] Add examples showing how to verify energy conservation with each spectrum type (Task 5)

Keeping open for remaining documentation tasks.
